### DISCUSSION TITLE
GSoC Idea: i18n improvements (crowd-sourcing)

### DISCUSSION THREAD

**Discussion by jmiranda** [openboxes/openboxes#2206] [Ideas]
### Overview 
OpenBoxes is available in any language / locale. However, we’ve only had the opportunity to translate to French and some Spanish. We’d like to provide a customizable database-backed message source as well as a way to integrate with a translation crowd-sourcing technology like phraseapp.

### Objectives
Provide runtime customizable translations using a database-backed i18n message source that can be optimized (cached). In addition, research and implement a crowd-sourcing solution to allow OpenBoxes to be translated into any language.

### Knowledge
* Spring
* i18n / Localization 
* Grails 3
* Translation Service (e.g. phraseapp, transifex)

### Resources
* https://phrase.com/blog/posts/database-stored-messages-for-i18n-in-spring-boot/
* https://docs.transifex.com/localization-tips-workflows/crowdsourcing-translations
* https://discuss.openboxes.com/t/how-to-translate-the-ui-in-any-language/188/4

### Mentors
* Justin Miranda 
* Kelsey Nagel

**Comment by comradekingu** [comment]
@jmiranda Please use Weblate. It has the best terms, the better license, and is (somehow unfortunately) a lot better than any of the current alternatives.
It is used by GNU Health, and other similar outfits.

**Reply by jmiranda** [reply]
Timely endorsement. Thank you.

**Reply by comradekingu** [reply]
I can help set up shop if you add "kingu" as an admin after filling out https://hosted.weblate.org/hosting/
There are also commercial plans, and paid professional support available.

Self-hosting is also possible under the same auspices, and I guess not out of question.
